CMakeLists.txt

```diff
@@ -1,148 +1,126 @@
-IF(BIICODE)
-	ADD_BIICODE_TARGETS()
-	if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/resources)
-		file(COPY ${CMAKE_CURRENT_SOURCE_DIR}/resources DESTINATION ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
-    ENDIF()
-    RETURN()
-ENDIF(BIICODE)
-cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
-cmake_policy(VERSION 2.6)
-if(POLICY CMP0063)
-	cmake_policy(SET CMP0063 OLD)
-endif()
-
-project(tinyxml2)
-include(GNUInstallDirs)
+cmake_minimum_required(VERSION 3.16)

```

Comment:
Do you **require** this recent version? E.g. some ubuntu distros are still at 3.10 so sticking to that makes adoption easier: https://packages.ubuntu.com/bionic/cmake
Yes I know updating CMake is easy, but I would require a recent version just because :)

Reply:
I didn't want to set a version earlier than I could easily test. I have 3.16 installed on Ubuntu 20.04 LTS. If you want to lower the version, I would welcome some assistance setting up GHA to use an older version.

Reply:
Certainly the version cannot be lowered below 3.14 (I think) as it stands because I use the newer features of GNUInstallDirs.

Reply:
On GHA you are already using an action which supports version selection: `lukka/get-cmake@v3.19.2` I think it makes sense, to use the one you specified as the minimum.
Looks like 3.15 is indeed required, didn't know about TARGET_FILE_BASE_NAME and would have just use what was there before. ;)

> newer features of GNUInstallDirs to automatically configure the install() commands

I'm interested: What are they?

Reply:
> > newer features of GNUInstallDirs to automatically configure the install() commands
> 
> I'm interested: What are they?

You can just write:

```
install(TARGETS foo EXPORT foo_exports)
```

Instead of:

```
install(
    TARGETS foo EXPORT foo_exports
    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
)
```

Basically it uses the correct GNUInstallDirs variables to set the destinations. It also introduced the `TYPE` argument: `TYPE INCLUDE` is the same as writing `DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"`. It's all just much cleaner. See the following for docs:

https://cmake.org/cmake/help/latest/release/3.14.html#commands

https://cmake.org/cmake/help/latest/command/install.html#installing-targets

https://cmake.org/cmake/help/latest/command/install.html#installing-files

Reply:
> On GHA you are already using an action which supports version selection: `lukka/get-cmake@v3.19.2` I think it makes sense, to use the one you specified as the minimum.

It does support version selection... but only as "low" as 3.17 🙂

Reply:
Thanks for the info, that was new to me :)
Oh, I see. Then maybe pip? Or https://github.com/marketplace/actions/actions-setup-cmake

Reply:
ah, this doesn't also install ninja. I see no need for that here, do you? Just omit the generator and let CMake choose

Reply:
I wanted to make sure that on Windows it wouldn't try to use Visual Studio (ie. a multi-config generator). I just dropped the min version to 3.15 and added testing for both the latest CMake and 3.15. Seems there's an issue with Windows on 3.15.